modules/api/src/main/java/io/fluo/api/config/FluoConfiguration.java

```diff
@@ -349,101 +371,101 @@ public FluoConfiguration setObservers(List<ObserverConfiguration> observers) {
   }
 
   public void setTransactionRollbackTime(long time, TimeUnit tu) {
-    setProperty(TRANSACTION_ROLLBACK_TIME_PROP, tu.toMillis(time));
+    setProperty(TRANSACTION_ROLLBACK_TIME_PROP, verifyPositive(TRANSACTION_ROLLBACK_TIME_PROP, tu.toMillis(time)));
   }
   
   public long getTransactionRollbackTime() {
-    return getLong(TRANSACTION_ROLLBACK_TIME_PROP, TRANSACTION_ROLLBACK_TIME_DEFAULT);
+    return verifyPositive(TRANSACTION_ROLLBACK_TIME_PROP, getLong(TRANSACTION_ROLLBACK_TIME_PROP, TRANSACTION_ROLLBACK_TIME_DEFAULT));
   }
 
   public FluoConfiguration setWorkerInstances(int workerInstances) {
-    setProperty(WORKER_INSTANCES_PROP, workerInstances);
+    setProperty(WORKER_INSTANCES_PROP, verifyPositive(WORKER_INSTANCES_PROP, workerInstances));
     return this;
   }
 
   public int getWorkerInstances() {
-    return getInt(WORKER_INSTANCES_PROP, WORKER_INSTANCES_DEFAULT);
+    return verifyPositive(WORKER_INSTANCES_PROP, getInt(WORKER_INSTANCES_PROP, WORKER_INSTANCES_DEFAULT));
   }
   
   public FluoConfiguration setWorkerMaxMemory(int maxMemoryMB) {
-    setProperty(WORKER_MAX_MEMORY_MB_PROP, maxMemoryMB);
+    setProperty(WORKER_MAX_MEMORY_MB_PROP, verifyPositive(WORKER_MAX_MEMORY_MB_PROP, maxMemoryMB));
     return this;
   }
   
   public int getWorkerMaxMemory() {
-    return getInt(WORKER_MAX_MEMORY_MB_PROP, WORKER_MAX_MEMORY_MB_DEFAULT);
+    return verifyPositive(WORKER_MAX_MEMORY_MB_PROP, getInt(WORKER_MAX_MEMORY_MB_PROP, WORKER_MAX_MEMORY_MB_DEFAULT));
   }
   
   public FluoConfiguration setWorkerNumCores(int numCores) {
-    setProperty(WORKER_NUM_CORES_PROP, numCores);
+    setProperty(WORKER_NUM_CORES_PROP, verifyPositive(WORKER_NUM_CORES_PROP, numCores));
     return this;
   }
   
   public int getWorkerNumCores() {
-    return getInt(WORKER_NUM_CORES_PROP, WORKER_NUM_CORES_DEFAULT);
+    return verifyPositive(WORKER_NUM_CORES_PROP, getInt(WORKER_NUM_CORES_PROP, WORKER_NUM_CORES_DEFAULT));
   }
   
   public FluoConfiguration setLoaderThreads(int numThreads) {
-    setProperty(LOADER_NUM_THREADS_PROP, numThreads);
+    setProperty(LOADER_NUM_THREADS_PROP, verifyPositive(LOADER_NUM_THREADS_PROP, numThreads));
     return this;
   }
   
   public int getLoaderThreads() {
-    return getInt(LOADER_NUM_THREADS_PROP, LOADER_NUM_THREADS_DEFAULT);
+    return verifyPositive(LOADER_NUM_THREADS_PROP, getInt(LOADER_NUM_THREADS_PROP, LOADER_NUM_THREADS_DEFAULT));
   }
   
   public FluoConfiguration setLoaderQueueSize(int queueSize) {
-    setProperty(LOADER_QUEUE_SIZE_PROP, queueSize);
+    setProperty(LOADER_QUEUE_SIZE_PROP, verifyPositive(LOADER_QUEUE_SIZE_PROP, queueSize));
     return this;
   }
   
   public int getLoaderQueueSize() {
-    return getInt(LOADER_QUEUE_SIZE_PROP, LOADER_QUEUE_SIZE_DEFAULT);
+    return verifyPositive(LOADER_QUEUE_SIZE_PROP, getInt(LOADER_QUEUE_SIZE_PROP, LOADER_QUEUE_SIZE_DEFAULT));
```

Comment:
That's a really good idea.  It'll make the code a lot cleaner.  I'll make the change.